app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people: dict = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    res_list: list = [Person(pers["name"], pers["age"]) for pers in people]
+    for elem in people:
+        if "wife" in elem.keys() and elem["wife"] is not None:
+            for key in Person.people:
+                if key == elem["wife"]:
+                    Person.people[elem["name"]].wife = Person.people[key]
+                    break
+        elif "husband" in elem.keys() and elem["husband"] is not None:
+            for key in Person.people:
+                if key == elem["husband"]:
+                    Person.people[elem["name"]].husband = Person.people[key]
+                    break
```

Comment:
simplify this code up to 1 cycle 'for'
remember that 1 wife is related to 1 husband

Reply:
I done it